jormungandr/src/network/mod.rs

```diff
@@ -16,12 +16,7 @@ mod subscription;
 // Constants
 
 mod buffer_sizes {
-    // Size of chunks to split processing of chain pull streams.
-    // Apart from sizing data chunks for intercom messages, it also
-    // determines how many blocks will be requested per each GetBlocks request
-    // distributed between different peers.
-    //
-    // This may need to be made into a configuration parameter.
+    // Size of buffer for processing of header push/pull streams.
     pub const CHAIN_PULL: usize = 32;
```

Comment:
don't you want to have this the same value as the one in the `src/blockchain/mod.rs` ?

Reply:
Not necessarily; this is a (minimal) buffer size for the MPSC channel, and the new one determines how many headers the blockchain task processes before it drops out. This should be implemented better and take into account the epoch and leadership schedule, so that we first request all blocks till the end of the next epoch, pause the stream, then when the blocks are received, somehow get a wakeup in the task that was waiting for them and process more headers with full validation.

Reply:
ok